etc/docker/Dockerfile

```diff
@@ -0,0 +1,136 @@
+# Multi-stage Dockerfile
+# Stages:
+#   1) distro-deps   - Install OS-level toolchain and libraries (base for all other stages)
+#   2) source-deps   - Install source-built dependencies (LLVM 11) on top of distro-deps
+#   3) app-source-*  - Prepare application sources (from local context or remote repo)
+#   4) app-build-*   - Build the application from the chosen app-source-* stage
+#
+# Use `--target` at build time to select which final build stage to produce:
+#   docker build --target=app-build-local -t wire-sysio:local .
+#   docker build --target=app-build-repo  -t wire-sysio:repo  .
+
+# Stage 1: distro-deps
+# Base image with system packages required for building (compilers, cmake, Python, etc.)
+FROM ubuntu:24.04 AS distro-deps
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV APP_DIR=/wire/wire-sysio
+ENV BUILD_DIR="${APP_DIR}/build"
+
+
+# Core package tools and repository helpers
+RUN apt-get update
+RUN apt-get install -y \
+        lsb-release \
+        wget \
+        software-properties-common
+
+# Enable Python 3.12 PPA and refresh indexes
+RUN add-apt-repository ppa:deadsnakes/ppa -y
+RUN apt-get update
+
+# Build toolchain and development libraries needed by the application
+RUN apt-get install -y \
+        gnupg \
+        build-essential \
+        cmake \
+        curl \
+        git \
+        vim \
+        sudo \
+        doxygen \
+        libboost-all-dev \
+        libcurl4-openssl-dev \
+        libgmp-dev \
+        libssl-dev \
+        libusb-1.0-0-dev \
+        pkg-config \
+        python3 \
+        zlib1g-dev \
+        libbz2-dev \
+        liblzma-dev \
+        libncurses5-dev \
+        libzstd-dev \
+    		python3.12
```

Comment:
I think we should remove `libboost-all-dev` from this list and below as boost is submoduled into Leap. Having it installed could cause confusion.
I would also add `jq` and `file` either here or in the ubuntu deps.